core/sdk-core/src/main/java/software/amazon/awssdk/core/exception/Ec2MetadataClientException.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.exception;
+
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Extends {@link SdkClientException} for EC2 Instance Metadata Service (IMDS) non-successful
+ * responses (4XX codes). Includes HTTP status codes to enable detailed error handling.
+ * <p>The status code is accessible via {@link #statusCode()} for specific error handling.
+ */
+
+@SdkPublicApi
+public class Ec2MetadataClientException extends SdkClientException {
```

Comment:
Should we move this to imds module? Classes should be `final` if they don't intend to be inherited